wallet/local/accounts.go

```diff
@@ -0,0 +1,219 @@
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary"
+
+	"github.com/ava-labs/avalanche-tooling-sdk-go/account"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/network"
+	"github.com/ava-labs/avalanche-tooling-sdk-go/utils"
+)
+
+// generateAccountName generates an automatic account name based on the current count
+func (w *LocalWallet) generateAccountName() string {
+	return fmt.Sprintf("account-%d", len(w.accounts)+1)
+}
+
+// setWalletAccount loads an account into the underlying primary wallet
+func (w *LocalWallet) setWalletAccount(ctx context.Context, acc account.Account, net network.Network) error {
+	wallet, err := getWalletFromAccount(ctx, acc, net, nil)
+	if err != nil {
+		return err
+	}
+	w.wallet = wallet
+	return nil
+}
+
+// Accounts returns all accounts managed by this wallet with their info
+func (w *LocalWallet) Accounts() map[string]account.AccountInfo {
+	result := make(map[string]account.AccountInfo)
+	for name := range w.accounts {
+		info, _ := w.Account(name)
+		result[name] = info
+	}
+	return result
+}
+
+// CreateAccount creates a new account with an optional name
+func (w *LocalWallet) CreateAccount(name string) (account.AccountInfo, error) {
+	// Generate automatic name if empty
+	if name == "" {
+		name = w.generateAccountName()
+	}
+
+	// Check if name already exists
+	if _, exists := w.accounts[name]; exists {
+		return account.AccountInfo{}, fmt.Errorf("account with name %q already exists", name)
+	}
+
+	// Create new local account
+	newAccount, err := account.NewLocalAccount()
+	if err != nil {
+		return account.AccountInfo{}, fmt.Errorf("failed to create new account: %w", err)
+	}
+
+	// Store in map
+	w.accounts[name] = newAccount
+
+	// Set as active if it's the first account
+	if len(w.accounts) == 1 {
+		if err := w.SetActiveAccount(name); err != nil {
+			return account.AccountInfo{}, fmt.Errorf("failed to set active account: %w", err)
+		}
+	}
+
+	// Return account info
+	return w.Account(name)
+}
+
+// Account returns info for a specific account by name
+func (w *LocalWallet) Account(name string) (account.AccountInfo, error) {
+	acc, exists := w.accounts[name]
+	if !exists {
+		return account.AccountInfo{}, fmt.Errorf("account %q not found", name)
+	}
+
+	pAddr, err := acc.GetPChainAddress(w.defaultNetwork)
+	if err != nil {
+		return account.AccountInfo{}, err
+	}
+
+	xAddr, err := acc.GetXChainAddress(w.defaultNetwork)
+	if err != nil {
+		return account.AccountInfo{}, err
+	}
+
+	cAddr, err := acc.GetCChainAddress()
+	if err != nil {
+		return account.AccountInfo{}, err
+	}
+
+	evmAddr, err := acc.GetEVMAddress()
+	if err != nil {
+		return account.AccountInfo{}, err
+	}
+
+	return account.AccountInfo{
+		Name:       name,
+		PAddress:   pAddr,
+		XAddress:   xAddr,
+		CAddress:   cAddr,
+		EVMAddress: evmAddr,
+	}, nil
+}
+
+// ImportAccount imports an account with a name
+func (w *LocalWallet) ImportAccount(name string, spec account.AccountSpec) (account.AccountInfo, error) {
```

Comment:
The use of account.AccountSpec may seem more simple and more convenient to import accounts quickly, but I think the additional struct of account.AccountSpec adds some confusion/bloat. To import an account, I believe we should have this sequence of steps which seems natural.

account := NewAccount("acct_name") // we may have other constructors like NewAccountFromPrivKey(privKey)
wallet.ImportAccount(account)

This simplicity keeps all Account constructors, getters, and setters functions located with the Account implementation  (/account/local_account.go)